### PR TITLE
Update express checkout section UI

### LINKED ISSUE #2393: Change Express checkouts section in admin
Link payment method is displayed in "Payment methods" . As this is an express checkout payment method e need to:
- Move Link payment method from "Payment methods" section to " Express checkouts"

We also need to change the "Express checkouts " section to have the UI similar to 
wcpay_express_checkouts

Context: p1658843212415559-slack-CHG7MTCAF

Additional changes:
* Update the “learn more” link that links to an express checkout section in the documentation where merchants can easily find information on Apple Pay, Google Pay, and Link
* In Link subsection include a hyperlink to the[ Link docs support article](https://woocommerce.com/document/stripe/#section-11)
* In Link subsection, include copy that state the “new payment experience (UPE) needs to be enabled for Link”
* Change “Stripe Link” to “Link by Stripe”
* Add description under Apple Pay / Google Pay – “Boost sales by offering a fast, simple, and secure checkout experience”
* Under Link - add description that includes ““Link autofills your customers’ payment and shipping details to deliver an easy and seamless checkout experience. By enabling this feature, you agree to the Link by Stripe terms and privacy policy.“

### PR DESCRIPTION
**Pull request by mordeth**
Fixes #2393

## Changes proposed in this Pull Request:
This PR updates the Express Checkout section in the Stripe payment method settings.

Context: p1658843212415559-slack-CHG7MTCAF

Screenshot 2022-10-26 at 17 34 42

Screenshot 2022-10-26 at 17 43 48

## Testing instructions
- Navigate to `WooCommerce` -> `Settings` -> `Payments` -> `Stripe`.
- Verify the Apple & Google Pay and Stripe Link are displayed in a separate section, `Express Checkouts`.
- Verify Stripe Link is available only when the Credit / Debit Card is activated.
- Verify enabling/disabling payment methods are fully functional on both the back-end and front-end.
- Navigate to `Apple & Google Pay` -> `Customize`.
- Verify the `Enable Apple Pay / Google Pay` is displayed in a separate section.
- Verify all the settings are saved and fully functional on both the back-end and front-end.

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [x] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

### CODE REVIEW

File: client/settings/payment-methods/index.js [hunk 1]
```diff
@@ -47,22 +37,17 @@ const PaymentMethodsDescription = () => {
...
 		<p>
 			{ __(
 				'Let your customers use their favorite express payment methods and digital wallets for faster, more secure checkouts across different parts of your store.',
 				'woocommerce-gateway-stripe'
 			) }
 		</p>
 		<ExternalLink href="https://woocommerce.com/document/stripe/#payment-request-buttons">
-			{ __( 'Learn more', 'woocommerce-gateway-stripe' ) }
+			{ __(
+				'Learn more(opens in a new tab)',
```
Comment (ricardo) [inline client/settings/payment-methods/index.js]: - This link should be just `Learn more`. The `(opens in a new tab)` part is a bug in WCPay: [#4033](https://github.com/Automattic/woocommerce-payments/pull/4033).
- I guess we should update the link to be https://woocommerce.com/document/stripe/#express-checkouts - Can you also update the documentation with new screenshots when this is released?

File: client/entrypoints/payment-request-settings/payment-request-page.js [hunk 2]
```diff
@@ -1,53 +1,52 @@
...
+			<PaymentRequestIcon size="medium" />
+		</div>
 		<p>
 			{ __(
-				'Decide how buttons for digital wallets Apple Pay and ' +
-					'Google Pay are displayed in your store. Depending on ' +
-					'their web browser and their wallet configurations, ' +
-					'your customers will see either Apple Pay or Google Pay, ' +
-					'but not both.',
+				'Allow your customers to collect payments via WooPay.',
```
Comment (ricardo) [inline client/entrypoints/payment-request-settings/payment-request-page.js]: > via WooPay.

The description doesn't sound right 😄.

File: client/settings/payment-request-section/index.js [hunk 3]
```diff
@@ -1,178 +1,205 @@
...
+											'woocommerce-gateway-stripe'
+										),
+										components: {
+											linkDocs: (
+												<a
+													target="_blank"
+													rel="noreferrer"
+													/* eslint-disable-next-line max-len */
+													href="https://woocommerce.com/document/payments/woocommerce-payments-stripe-link/"
+												/>
```
Comment (ricardo) [inline client/settings/payment-request-section/index.js:190]: This link is pointing to the WCPay documentation. Should it be `https://woocommerce.com/document/stripe/#stripe-link` ?

File: client/settings/payment-request-section/index.js [hunk 4]
```diff
@@ -1,178 +1,205 @@
...
+						</div>
+					</li>
+					{ displayLinkPaymentMethod && (
+						<li className="express-checkout has-icon-border">
+							<div className="express-checkout__checkbox loadable-checkbox label-hidden">
+								<CheckboxControl
+									label={ __(
+										'Link by Stripe Input',
+										'woocommerce-payments'
+									) }
```
Comment (ricardo) [inline client/settings/payment-request-section/index.js:127]: 🤏 small leftover on `woocommerce-payments`.

File: client/settings/payment-request-section/index.js [hunk 5]
```diff
@@ -1,178 +1,205 @@
...
+			updateEnabledMethodIds( [
+				...new Set( [ ...enabledMethodIds, 'link' ] ),
 			] );
 		} else {
-			updatePaymentRequestLocations(
-				paymentRequestLocations.filter( ( name ) => name !== location )
-			);
+			updateEnabledMethodIds( [
+				...enabledMethodIds.filter( ( id ) => id !== 'link' ),
+			] );
```
Comment (ricardo) [inline client/settings/payment-request-section/index.js]: I think some parts of this function might be too verbose. What do you think of simplifying this to: 👇 ?

```javascript
const toggleStripeLink = ( isEnabled ) => {
	// Add/remove Stripe Link from the list of enabled payment methods.
	if ( isEnabled ) {
		updateEnabledMethodIds( [ ...enabledMethodIds, 'link' ] );
	} else {
		updateEnabledMethodIds( [
			...enabledMethodIds.filter( ( id ) => id !== 'link' ),
		] );
	}
};
```

File: client/settings/payment-request-section/index.js [hunk 6]
```diff
@@ -1,178 +1,205 @@
...
+									) }
+								</div>
+								<div className="express-checkout__description">
+									{
+										/* eslint-disable jsx-a11y/anchor-has-content */
+										interpolateComponents( {
+											mixedString: __(
+												'Link autofills your customers’ payment and shipping details to ' +
+													'deliver an easy and seamless checkout experience. ' +
+													'New payment experience (UPE) needs to be enabled for Link. ' +
```
Comment (ricardo) [inline client/settings/payment-request-section/index.js]: I think this is called ["New checkout experience"](https://woocommerce.com/document/stripe/#new-checkout-experience) in Stripe, and users might not be familiar with the term UPE 🤔.

File: client/settings/payment-request-section/index.js [hunk 7]
```diff
@@ -1,178 +1,205 @@
...
+												),
+											},
+										} )
+										/* eslint-enable jsx-a11y/anchor-has-content */
+									}
+								</div>
+							</div>
+							<div className="express-checkout__link">
+								{
+									/* eslint-disable jsx-a11y/anchor-has-content */
```
Comment (ricardo) [inline client/settings/payment-request-section/index.js:178]: 🤏 These `eslint-disable` rules might not be necessary for this repository.

File: client/settings/payment-request-section/styles.scss [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+				padding-left: 16px;
+
+				a {
+					white-space: nowrap;
+				}
+			}
+
+			.components-checkbox-control__label {
+				display: none;
+			}
```
Comment (ricardo) [inline client/settings/payment-request-section/styles.scss:70]: This section looks a bit boxed up on mobile.

If we add this to the label container, it looks more like the Payment methods section:

```css
&__label-container {
	flex: 1;
}
```

| Before | After |
| -- | -- |
| Express checkouts section on mobile CSS Before | Express checkouts section on mobile CSS After |

File: client/entrypoints/payment-request-settings/__tests__/payment-request-button-preview.test.js [hunk 9]
```diff
@@ -25,6 +25,7 @@ jest.mock( '@stripe/react-stripe-js', () => ( {
 } ) );
 
 jest.mock( 'wcstripe/data', () => ( {
+	usePaymentRequestEnabledSettings: jest.fn(),
```
Comment (ricardo) [inline client/entrypoints/payment-request-settings/__tests__/payment-request-button-preview.test.js]: Does this make a difference in the JS tests execution?

**Review by ricardo — CHANGES_REQUESTED** [review]
Tests well. Just left a few minor comments.

File: client/settings/payment-methods/index.js (hunk 1 above)
Comment (mordeth) [reply client/settings/payment-methods/index.js]: Updated.

File: client/entrypoints/payment-request-settings/payment-request-page.js (hunk 2 above)
Comment (mordeth) [reply client/entrypoints/payment-request-settings/payment-request-page.js]: Updated. :)

File: client/settings/payment-request-section/index.js (hunk 3 above)
Comment (mordeth) [reply client/settings/payment-request-section/index.js:190]: Updated.

File: client/settings/payment-request-section/index.js (hunk 4 above)
Comment (mordeth) [reply client/settings/payment-request-section/index.js:127]: Fixed.

File: client/settings/payment-request-section/index.js (hunk 5 above)
Comment (mordeth) [reply client/settings/payment-request-section/index.js]: Looks much better, updated.

File: client/settings/payment-request-section/index.js (hunk 6 above)
Comment (mordeth) [reply client/settings/payment-request-section/index.js]: Makes sense, updated.

File: client/settings/payment-request-section/index.js (hunk 7 above)
Comment (mordeth) [reply client/settings/payment-request-section/index.js:178]: Actually, tried removing it, and JS lint errors are thrown. We have to keep them.

File: client/settings/payment-request-section/styles.scss (hunk 8 above)
Comment (mordeth) [reply client/settings/payment-request-section/styles.scss:70]: Good catch, updated!

File: client/entrypoints/payment-request-settings/__tests__/payment-request-button-preview.test.js (hunk 9 above)
Comment (mordeth) [reply client/entrypoints/payment-request-settings/__tests__/payment-request-button-preview.test.js]: Good catch, `usePaymentRequestEnabledSettings ` is not used in those tests.

**Review by dpaun1985 — APPROVED** [review]
✅ LGTM
Tested and work as expected!

File: client/settings/payment-request-section/index.js [hunk 10]
```diff
@@ -1,178 +1,199 @@
...
+	const updateStripeLinkCheckout = ( isEnabled ) => {
+		// Add/remove Stripe Link from the list of enabled payment methods.
+		if ( isEnabled ) {
+			updateEnabledMethodIds( [ ...enabledMethodIds, 'link' ] );
 		} else {
-			updatePaymentRequestLocations(
-				paymentRequestLocations.filter( ( name ) => name !== location )
-			);
+			updateEnabledMethodIds( [
+				...enabledMethodIds.filter( ( id ) => id !== 'link' ),
```
Comment (mgascam) [inline client/settings/payment-request-section/index.js]: This is minor, but I see we are using 'link' in several places in this file, maybe we could move it into a constant?

File: client/settings/payment-request-section/index.js [hunk 11]
```diff
@@ -1,178 +1,199 @@
...
-					<h4>
-						{ __(
-							'Show express checkouts on',
-							'woocommerce-gateway-stripe'
-						) }
-					</h4>
-					<ul>
-						<li>
+		<Card className="express-checkouts">
+			<CardBody size={ 0 }>
```
Comment (mgascam) [inline client/settings/payment-request-section/index.js]: Can you share more context about why we need to pass 0 `size`? For clarification, I would not expect to pass 0

File: client/settings/payment-request-section/index.js [hunk 12]
```diff
@@ -1,178 +1,199 @@
...
-							<CheckboxControl
-								disabled={ ! isPaymentRequestEnabled }
-								checked={
-									isPaymentRequestEnabled &&
-									paymentRequestLocations.includes( 'cart' )
+							</div>
+							<div className="express-checkout__description">
+								{
+									/* eslint-disable jsx-a11y/anchor-has-content */
+									interpolateComponents( {
```
Comment (mgascam) [inline client/settings/payment-request-section/index.js:76]: TIL

**Review by mgascam — APPROVED** [review]
LGTM! I just left some minor comments, feel free to skip them.

I followed testing instructions and works as expected.

**Review by ricardo — APPROVED** [review]
LGTM. Tests well.

@mordeth I noticed we don't have the "Requires currency" label anymore in the express checkouts section. Should we add that in a different issue?

stripe Link requires currency

Also, do we have any issues/notes on updating the screenshots on the [express checkouts](https://woocommerce.com/document/stripe/#express-checkouts) documentation once this is released?

File: client/settings/payment-request-section/index.js (hunk 10 above)
Comment (mordeth) [reply client/settings/payment-request-section/index.js]: Makes sense, done!

File: client/settings/payment-request-section/index.js (hunk 11 above)
Comment (mordeth) [reply client/settings/payment-request-section/index.js]: You are right, we do not need size at all.

**Comment by kalessil** [comment]
@mordeth, thanks for working on this PR! Can you also respond to Ricardo's questions above? Thanks!

**Comment by mordeth** [comment]
@ricardo, I will create an issue for the `Requires currency`, and I have requested a docs update.

**Pull request merged**